src/Groups/homomorphisms.jl

```diff
@@ -542,6 +542,14 @@ function isomorphism(::Type{FPGroup}, G::GAPGroup; on_gens::Bool=false)
          f = GAP.Globals.GroupHomomorphismByImages(G.X, GAP.Globals.FreeGroup(0), GAP.Obj([]), GAP.Obj([]))
          GAP.Globals.SetIsBijective(f, true)
        else
+         # The computations are easy if `Ggens` is a pcgs,
+         # otherwise GAP will call `CoKernel`.
+         if GAP.Globals.HasFamilyPcgs(G.X)
+           pcgs = GAP.Globals.InducedPcgsWrtFamilyPcgs(G.X)
+           if pcgs == Ggens
+             Ggens = pcgs
```

Comment:
I don't understand: if they are equal, why does this have an effect?


Or is one a pcgs object, the other a plain list, and they happen to compare as equal? If so, this is a subtle point and I think a comment should explain it?

Reply:
Yes, `pcgs` is an object in the filter `IsPcgs`, whereas `Ggens` is a plain list.